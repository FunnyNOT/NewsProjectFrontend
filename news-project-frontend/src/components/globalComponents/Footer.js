import React from 'react'

function Footer() {
  return (
    <div className='footer'>
      <div className='footer__container'>
        <div className='footer__container__left'>
          <div className='footer__container__left__logo'>
            <img src={`${process.env.PUBLIC_URL}/images/logo.png`} alt='Logo' height='56' />
          </div>
          <div className='footer__container__left__text'>
            <p>Â© 2024 News Project</p>
          </div>
        </div>
        <div className='footer__container__right'>
          <div className='footer__container__right__text'>
            <p>
              Created by: <a href=''>News Project</a>
            </p>
          </div>
        </div>
      </div>
    </div>
  )
}

export { Footer }
