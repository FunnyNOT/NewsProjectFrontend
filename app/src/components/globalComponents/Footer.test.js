// src/__tests__/Footer.test.js
import React from 'react'
import { render } from '@testing-library/react'
import { Footer } from './Footer'

test('renders footer content', () => {
  const { getByText, getByAltText } = render(<Footer />)

  // Assert the presence of specific text content
  expect(getByText('Â© 2024 News Project')).toBeInTheDocument()
  expect(getByText('Created by:')).toBeInTheDocument()
  expect(getByText('News Project')).toBeInTheDocument()

  // Assert the presence of the logo (modify the alt text as needed)
  expect(getByAltText('Logo')).toBeInTheDocument()
})

test('contains the correct link', () => {
  const { getByText } = render(<Footer />)

  // Assert that the link text is present and contains an href attribute
  const linkElement = getByText('News Project')
  expect(linkElement).toBeInTheDocument()
  expect(linkElement.closest('a')).toHaveAttribute('href')
})

test('renders correctly', () => {
  const { container } = render(<Footer />)
  expect(container).toMatchSnapshot()
})
