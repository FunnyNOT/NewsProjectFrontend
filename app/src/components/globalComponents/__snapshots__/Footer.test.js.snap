// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class="footer__container"
    >
      <div
        class="footer__container__left"
      >
        <div
          class="footer__container__left__logo"
        >
          <img
            alt="Logo"
            height="56"
            src="/images/logo.png"
          />
        </div>
        <div
          class="footer__container__left__text"
        >
          <p>
            Â© 2024 News Project
          </p>
        </div>
      </div>
      <div
        class="footer__container__right"
      >
        <div
          class="footer__container__right__text"
        >
          <p>
            Created by: 
            <a
              href=""
            >
              News Project
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
